library/manageiq_custom_attributes.py

```diff
@@ -150,6 +150,15 @@ def update_custom_attribute(self, entity_type, entity_id, ca, ca_href):
         except Exception as e:
             self.module.fail_json(msg="Failed to update the custom attribute {ca_name}. Error: {error}".format(ca_name=ca['name'], error=e))
 
+    @staticmethod
+    def compare_custom_attributes(ca1, ca2):
+        if 'section' in ca1 and 'section' in ca2:
```

Comment:
I'm not sure this logic is right.
If only one of them has a section, the other's section is ignored and we skip to comparing names?

I'd expect something like
```
return (ca1.get('section'), ca1['name']) == (ca2.get('section'), ca2['name'])
```